lib/scala/text-buffer/src/main/scala/org/enso/text/editing/model.scala

```diff
@@ -58,4 +60,19 @@ object model {
     */
   case class TextEdit(range: Range, text: String)
 
+  /** A text span denoted by its absolute start and end indexes.
+    *
+    * @param start the start index of a span
+    * @param end the end index of a span
+    */
+  case class Span(start: Int, end: Int) {
+    def length: Int = end - start
+  }
+
+  /** The mapping between the text spans and their identifiers.
+    *
+    * @param values the list of span-uuid pairs
+    */
+  case class IdMap(values: Vector[(Span, UUID)])
```

Comment:
Why do we have `Pair` of `Pair` and `UUID` when the protocol talkes about **tripple**?